backend/aethel_db/views.py

```diff
@@ -0,0 +1,119 @@
+from dataclasses import asdict, dataclass, field
+from typing import List, Optional
+from django.http import HttpRequest, JsonResponse
+from django.conf import settings
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from aethel import ProofBank
+from aethel.frontend import Sample, LexicalItem
+from aethel_db.filters import in_lemma, in_word
+from parseport.logger import logger
+from src.aethel.scripts.search import search
+
+FULL_DATASET_PATH = getattr(settings, "FULL_DATASET_PATH")
+DATA_SUBSET_PATH = getattr(settings, "DATA_SUBSET_PATH")
+
+if getattr(settings, "DEBUG", False) is False:
+    dataset = ProofBank.load_data(FULL_DATASET_PATH)
+else:
+    try:
+        dataset = ProofBank.load_data(DATA_SUBSET_PATH)
+    except FileNotFoundError:
+        logger.error(
+            "Aethel subset not found. Have you run create_aethel_subset? Using full dataset instead."
+        )
+        dataset = ProofBank.load_data(FULL_DATASET_PATH)
+
+
+@dataclass
+class AethelSample:
+    name: str
+    sentence: str
+
+
+@dataclass
+class AethelListItem:
+    lemma: str
+    word: str
+    type: str
+    samples: List[AethelSample] = field(default_factory=list)
+
+
+@dataclass
+class AethelListResponse:
+    """
+    Response object for Aethel query view.
+    """
+
+    results: List[AethelListItem] = field(default_factory=list)
+    error: Optional[str] = None
+
+    def existing_result(
+        self, lemma: str, word: str, type: str
+    ) -> Optional[AethelListItem]:
+        """
+        Return an existing result with the same lemma, word, and type, if it exists.
+        """
+        for item in self.results:
```

Comment:
probably premature optimization but, if you use a dictionary indexed by tuples of ```(lemma, word, type)``` you can avoid the for loop

Reply:
Ah, very nice! Good to keep in mind if we run into performance issues.